gaurav/src/pages/recipe/RecipeDetail.tsx

```diff
@@ -0,0 +1,102 @@
+import { Loader } from "lucide-react";
+import { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { IRecipeData } from "@/lib/interfaces";
+import { apiClient } from "@/api/apiClient";
+
+export const RecipeDetail = () => {
+  const { id } = useParams<{ id: string }>();
+  const [recipe, setRecipe] = useState<IRecipeData | null>(null);
+  const [timeout, setTimeOut] = useState<boolean>(false);
+
+  const fetchRecipeDetail = async (id?: string) => {
+    try {
+      const res = await apiClient.get(`lookup.php?i=${id}`);
+      const data = res.data.meals[0];
+      setRecipe(data);
+    } catch (error) {
+      console.log("Error", error);
+    }
+  };
+  useEffect(() => {
+    setTimeout(() => {
+      setTimeOut(false);
+    }, 1000);
+    if (id) {
+      fetchRecipeDetail(id);
+    }
+    setTimeOut(false);
+  }, [id]);
+
+  console.log(recipe);
+  if (!recipe) {
+    return (
+      <div className="h-screen bg-recipebg text-white cursor-wait flex items-center justify-center">
+        Loading <Loader className="animate-spin w-5 h-5 mx-2" />
+      </div>
+    );
+  }
+  const ingredients = [];
+  for (let i = 1; i <= 20; i++) {
+    if (recipe[`strIngredient${i}`]) {
+      ingredients.push(`${recipe[`strIngredient${i}`]} - ${recipe[`strMeasure${i}`]}`);
+    }
+  }
+  const steps = recipe.strInstructions.split(/(?:STEP\s*\d*[:.]?\s*)/i).filter((step) => step.trim() !== "");
```

Comment:
what does this do?

Reply:
this  splits the given api single response string into different steps
ex - `STEP 1\r\nTO MAKE SUSHI ROLLS: Pat out some rice. Lay a nori sheet on the mat, shiny-side down. Dip your hands in the vinegared water, then pat handfuls of rice on top in a 1cm thick layer, leaving the furthest edge from you clear.\r\n\r\nSTEP 2\r\nSpread over some Japanese mayonnaise. Use a spoon to spread out a thin layer of mayonnaise down the middle of the rice.` into separate steps